discovery/openstack/openstack.go

```diff
@@ -91,12 +94,19 @@ func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err != nil {
 		return err
 	}
+
+	availability := gophercloud.Availability(c.Availability)
+	if availability != gophercloud.AvailabilityPublic && availability != gophercloud.AvailabilityInternal && availability != gophercloud.AvailabilityAdmin {
```

Comment:
Do we have to hardcode this here, or is there a function we can use? Is it likely that this will change?

Reply:
There is no function in the gophercloud package at least. this is very unlikely to change – it denotes wether the api endpoint is reachable for the public or only within the cloud overlay of the openstack deployment.

Reply:
Alternatively, we can skip validation here, and allow any string. In my opinion it is nice though to protect against typos in the configuration validation already, instead of having trouble to investigate follow up errors (no endpoint found). I don't see a use case for more values than the defined constants in gophercloud

Reply:
Okay then. This would be cleaner with a select and using plain strings.

Reply:
I tried to incorporate your suggestions. is it fine now?